internal/storage/storage.go

```diff
@@ -0,0 +1,177 @@
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/flynntdev/go_final_project/api"
+	"github.com/flynntdev/go_final_project/config"
+	"github.com/flynntdev/go_final_project/internal/task"
+)
+
+const (
+	limit = 20
+)
+
+type Store struct {
+	db *sql.DB
+}
+
+func NewStore(db *sql.DB) Store {
+	return Store{db: db}
+}
+
+func (s *Store) PostTask(t task.Task) (string, error) {
+	var err error
+	err = t.CheckTitle()
+	if err != nil {
+		return "", err
+	}
+	parseDate, err := t.CheckData()
+	if err != nil {
+		return "", err
+	}
+	t.Date, err = t.CheckRepeat(parseDate)
+	if err != nil {
+		return "", err
+	}
+	query := `INSERT INTO scheduler (date, title, comment, repeat) VALUES (?, ?, ?, ?)`
+	res, err := s.db.Exec(query, t.Date, t.Title, t.Comment, t.Repeat)
+	if err != nil {
+		return "", fmt.Errorf(`{"error":"Ошибка добавления задачи в базу данных"}`)
```

Comment:
Теряешь текст ошибки. Тут ты формируешь сразу ответ в АПИ, но это слой хранения и тут должна быть максимально информативная ошибка
```suggestion
		return "", fmt.Errorf("error in executing query: %w", err)
```

Reply:
Это прям во всех местах у тебя, нужно все править